### PR TITLE
[follow-up] Add tests verifying that alerts expressions are valid

### PR DESCRIPTION
**Pull request by rexagod**
Follows up on #26787.

### CODE REVIEW

**Comment by rexagod** [comment]
```
github.com/openshift/origin/test imports
	k8s.io/kubernetes/pkg/registry/registrytest imports
	k8s.io/apiserver/pkg/registry/generic/registry imports
	k8s.io/apiserver/pkg/storage/storagebackend/factory imports
	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc imports
	go.opentelemetry.io/otel/metric/global: module go.opentelemetry.io/otel/metric@latest found (v1.16.0), but does not contain package go.opentelemetry.io/otel/metric/global
github.com/openshift/origin/test imports
	k8s.io/kubernetes/pkg/registry/registrytest imports
	k8s.io/apiserver/pkg/registry/generic/registry imports
	k8s.io/apiserver/pkg/storage/storagebackend/factory imports
	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc imports
	go.opentelemetry.io/otel/metric/instrument: module go.opentelemetry.io/otel/metric@latest found (v1.16.0), but does not contain package go.opentelemetry.io/otel/metric/instrument
```

[otel/metric](https://pkg.go.dev/go.opentelemetry.io/otel/metric) does not have either of those packages currently. I'll try updating the package, or, in the worst case scenario, add replace directives to point to wherever these packages were moved to.

File: go.mod [hunk 1]
```diff
@@ -279,6 +287,7 @@ require (
 
 replace (
 	github.com/onsi/ginkgo/v2 => github.com/openshift/onsi-ginkgo/v2 v2.6.1-0.20230317131656-c62d9de5a460
+	go.opentelemetry.io/otel/metric => go.opentelemetry.io/otel/metric v0.38.1
```
Comment (rexagod) [inline go.mod:290]: pPointed this to the latest version thta statifies the issues stemming from .

**Comment by rexagod** [comment]
Unholded for now, just want to see how checks are doing. This is still very much a WIP.

File: go.mod (hunk 1 above)
Comment (rexagod) [reply go.mod:290]: Mentioning here that the parent PR is free of these dependency conflicts, which come into play once that's rebased on `master`.

Comment (rexagod) [reply go.mod:290]: https://github.com/open-telemetry/opentelemetry-go/issues/4255

Comment (rexagod) [reply go.mod]: `go vet ./...` and `./hack/update-generated` passing now.

Comment (rexagod) [reply go.mod]: `go vet ./...` and `./hack/update-generated.sh` passing now.

File: test/extended/prometheus/prometheus.go [hunk 2]
```diff
@@ -257,6 +361,95 @@ var _ = g.Describe("[sig-instrumentation][Late] OpenShift alerting rules [apigro
...
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			if len(resp.Data.Result) > 0 {
+				return sets.NewString("one of the metrics used in the query has invalid label(s)")
+			}
+
+			return nil
+		})
+		if err != nil {
+			testresult.Flakef(err.Error())
```
Comment (rexagod) [inline test/extended/prometheus/prometheus.go:450]: .

File: test/extended/prometheus/prometheus.go [hunk 3]
```diff
@@ -105,6 +107,108 @@ var _ = g.Describe("[sig-instrumentation][Late] OpenShift alerting rules [apigro
 		"ClusterVersionOperatorDown",
 	)
 
+	// These alerts are known to have metrics that may be absent or metric
+	// selectors returning empty results. In most cases, these are errors or
+	// conditional metrics which are only present if an error or a certain
+	// condition occurs. In these cases, we can skip the validation of the
+	// alert since we won't be able to easily tell the difference between what
+	// is expected and what isn't.
+	transientAlertExceptions := sets.NewString(
```
Comment (simonpasquier) [inline test/extended/prometheus/prometheus.go:116]: Given that we have so many exceptions, I'm not convinced by the usefulness of the test. I fear that we'll be constantly bugged to add new alerting rules to this list...

Comment (rexagod) [reply test/extended/prometheus/prometheus.go:116]: ACK, should I close this PR (and only validate the collection-based metrics on CMO side)?

Comment (simonpasquier) [reply test/extended/prometheus/prometheus.go:116]: I think so :bow:

**Pull request closed without merging**